nevergrad/functions/pbo/core.py

```diff
@@ -0,0 +1,121 @@
+    base_function: str
+        Which base function to use. Either "OneMax" or "LeadingOnes"
+    iid: int
+        the instance of the function, specifies the transformations in variable and objective space
+    dim: int
+        The dimensionality of the problem.
+    dummy:
+        Float between 0 and 1, fractoin of valid bits.
+    epistasis:
+        size of sub-string for epistasis
+    neutrality:
+        size of sub-string for neutrality
+    ruggedness:
+        gamma for ruggedness layper
+    instrum_str: str
+        How the paramerterization is handled. Either "Softmax", in which case the optimizers will work on probabilities
+        (so the same parametrization may actually lead to different values), or "Unordered". With this one the optimizer 
+        will handle a value, which will lead to a 0 if it is negative, and 1 if it is positive. 
+        This way it is no more stochastic, but the problem becomes very non-continuous
+        
+    Notes
+    -----
+    The IOHexperimenter is part of the IOHprofiler project (IOHprofiler.github.io)
+
+    References
+    -----
+    For a description of the W-model in the IOHexperimenter and of the used transformation methods,
+    please see "Benchmarking discrete optimization heuristics with IOHprofiler" by C. Doerr et al.
+    """
+
+    def __init__(self, base_function: str = "OneMax", iid: int = 0, dim: int = 16, dummy: float = 0, epistasis: int = 0,
+                 neutrality: int = 0, ruggedness: int = 0, instrum_str: str = "Softmax") -> None:
+        assert epistasis <= dim, "Epistasis has to be less or equal to than dimension"
+        assert neutrality <= dim, "Neutrality has to be less than or equal to dimension"
+        assert ruggedness <= dim ** 2, "Ruggedness has to be less than or equal to dimension squared"
+        assert dummy <= 1 and dummy >= 0, "Dummy variable fraction has to be in [0,1]"
+        
+        if base_function == "OneMax":
+            self.f_internal = W_Model_OneMax(iid, dim)
+        elif base_function == "LeadingOnes":
+            self.f_internal = W_Model_LeadingOnes(iid, dim)
+        self.f_internal.set_w_setting(dummy, epistasis, neutrality, ruggedness)
+        assert instrum_str in ["Softmax", "Unordered"], "The only valid options for 'instrum_str' are 'Softmax' and 'Unordered'"
+        if instrum_str == "Softmax":
+            parameterization = ng.p.Choice([0, 1], repetitions=dim)
+        else:
+            parameterization = ng.p.TransitionChoice([0, 1], repetitions=dim)
+        super().__init__(self._evaluation_internal, parameterization)
+        self.descriptors.update(base_function = base_function, iid = iid, dummy = dummy, epistasis = epistasis, 
+                                neutrality = neutrality, ruggedness = ruggedness)
+        self.register_initialization()
+
+
+    def _evaluation_internal(self, x: np.ndarray) -> float:
+        assert len(x) == self.dimension, f"Expected dimension {self.dimension}, got {len(x)}"
```

Comment:
thinking about it, you can just remove this check, the problem is only that self.dimension does not correspond to the dimension of the input of the function when using Choice, but this is not an issue.

Reply:
... and it's the same for the check at line 55, right ?